modules/thermal_hydraulics/src/components/Component.C

```diff
@@ -158,6 +159,68 @@ Component::checkComponentExistsByName(const std::string & comp_name) const
     logError("The component '", comp_name, "' does not exist");
 }
 
+bool
+Component::addRelationshipManagersFromParameters(const InputParameters & moose_object_pars)
+{
+  const auto & buildable_types = moose_object_pars.getBuildableRelationshipManagerTypes();
+
+  bool added = false;
+
+  for (const auto & buildable_type : buildable_types)
+  {
+    auto & rm_name = std::get<0>(buildable_type);
+    auto & rm_type = std::get<1>(buildable_type);
+    auto rm_input_parameter_func = std::get<2>(buildable_type);
+
+    added = addRelationshipManager(moose_object_pars, rm_name, rm_type, rm_input_parameter_func) ||
```

Comment:
I don't know what you're talking about regarding 1st, 2nd, and 3rd types. Is this a comment about components specifically, or even the existing method in `Action`?

Reply:
What I mean is, are you saying it made sense in the context of actions but not in the context of components, or are you saying it never made sense?

Reply:
it seems odd in both context

Reply:
I added this in 42c3d733790b34adc95d41b6d634a58213568d2b. @joshuahansel I don't think you need the `added` logic here. I created it for a pretty specific purpose. In the `Action` itself it would probably be better to rename the local variable `any_added`

Reply:
Oh, did you see the `|| added` after the `addRelationshipManager()` call? So basically, `added` would be better named as `added_any`.

Reply:
Alex's last comment wasn't there when I responded - I guess we responded at nearly the same time :D

Reply:
> I don't think you need the added logic here.

I'll make the method return void then.